include/shark/Algorithms/Trainers/CSvmTrainer.h

```diff
@@ -251,6 +266,9 @@ class CSvmTrainer : public AbstractSvmTrainer<
 			typedef SvmShrinkingProblem<SVMProblemType> ProblemType;
 			ProblemType problem(svmProblem,base_type::m_shrinking);
 			QpSolver< ProblemType > solver(problem);
+			if (svm.alpha().size1() > 0 && svm.alpha().size2() == 1) {
+				problem.setInitialSolution(blas::column(svm.alpha(), 0));
```

Comment:
Hi. You should rather use some kind of equality that checks whether the pointers are the same and in the same order. I think op== does that
________________________________________
From: Tobias Glasmachers [notifications@github.com]
Sent: Friday, June 24, 2016 9:42 PM
To: Shark-ML/Shark
Cc: Oswin Krause; Comment
Subject: Re: [Shark-ML/Shark] Added support for warm-starts into the CSvmTrainer. (#86)

In include/shark/Algorithms/Trainers/CSvmTrainer.h<https://github.com/Shark-ML/Shark/pull/86#discussion_r68452029>:

> @@ -251,6 +266,9 @@ class CSvmTrainer : public AbstractSvmTrainer<
>                       typedef SvmShrinkingProblem<SVMProblemType> ProblemType;
>                       ProblemType problem(svmProblem,base_type::m_shrinking);
>                       QpSolver< ProblemType > solver(problem);
> +                     if (svm.alpha().size1() > 0 && svm.alpha().size2() == 1) {
> +                             problem.setInitialSolution(blas::column(svm.alpha(), 0));


What you hint at is a serious problem with the current interface. Changing C is relatively easy to deal with. We have to loop over the vector anyway, so checking the constraints is nearly for free. But what if the data set changes, i.e., the basis of the kernel expansion differs from the input portion of the LabeledData object passed to the train function? I could check for equality of the shared pointers inside the data containers - do we have any interface to check whether two data objects share data? We have data.makeIndependent(), but what I need here is data.isSharedWith(otherData). Any objections to adding such a function?

�
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/Shark-ML/Shark/pull/86/files/1d8bc64b7de5c63be11651a7af21ee36444340bd#r68452029>, or mute the thread<https://github.com/notifications/unsubscribe/AOWTBnoYtSKx_tnSaVUXI4UpoZvc9a0Yks5qPDMwgaJpZM4I8ua_>.